ctms/acoustic_service.py

```diff
@@ -288,22 +286,65 @@ def _add_contact(self, list_id=None, sync_fields=None, columns=None):
             columns = {}
         if sync_fields is None:
             sync_fields = {}
-        self.acoustic.add_recipient(
-            list_id=list_id,
-            created_from=3,
-            update_if_found="TRUE",
-            allow_html=False,
-            sync_fields=sync_fields,
-            columns=columns,
-        )
+        params = {
+            "list_id": list_id,
+            "created_from": 3,
+            "update_if_found": "TRUE",
+            "allow_html": False,
+            "sync_fields": sync_fields,
+            "columns": columns,
+        }
+        if self.metric_service:
+            start_time = time.monotonic()
+            err = None
+            status = "success"
+            try:
+                self.acoustic.add_recipient(**params)  # Call to Acoustic
+            except Exception as e:  # pylint: disable=broad-except
+                err = e
+                status = "failure"
+            duration = time.monotonic() - start_time
```

Comment:
I think the metrics stuff should be in a ``finally`` clause, so that exceptions raised by the metrics code doesn't hide exceptions from acoustic. Something like:

```python
except Exception as e:
  status = "failure"
  raise
finally:
  duration = time.monotonic() - start_time
  # The rest of the metrics code
``` 

I'm tempted to suggest a metrics context manager, but I think that would obfuscate otherwise straightforward code.